src/nuclide_header.F90

```diff
@@ -0,0 +1,755 @@
+      import NuclideMG
+      class(NuclideMG), intent(in) :: this
+      integer, intent(in)           :: gin   ! Incoming Energy Group
+      integer, intent(in)           :: gout  ! Outgoing Energy Group
+      real(8), intent(in)           :: mu    ! Angle of interest
+      real(8), intent(in), optional :: uvw(3) ! Direction vector
+      integer, intent(in), optional :: i_azi ! Incoming Energy Group
+      integer, intent(in), optional :: i_pol ! Outgoing Energy Group
+      real(8)                       :: f     ! Return value of f(mu)
+
+    end function nuclide_calc_f_
+  end interface
+
+!===============================================================================
+! NuclideIso contains the base MGXS data for a nuclide specifically for
+! isotropically weighted MGXS
+!===============================================================================
+
+  type, extends(NuclideMG) :: NuclideIso
+
+    ! Microscopic cross sections
+    real(8), allocatable :: total(:)        ! total cross section
+    real(8), allocatable :: absorption(:)   ! absorption cross section
+    real(8), allocatable :: scatter(:,:,:)  ! scattering information
+    real(8), allocatable :: nu_fission(:,:) ! fission matrix (Gout x Gin)
+    real(8), allocatable :: k_fission(:)    ! kappa-fission
+    real(8), allocatable :: fission(:)      ! neutron production
+    real(8), allocatable :: chi(:)          ! Fission Spectra
+    real(8), allocatable :: mult(:,:)       ! Scatter multiplicity (Gout x Gin)
+
+  contains
+    procedure :: print  => nuclideiso_print  ! Writes nuclide info
+    procedure :: get_xs => nuclideiso_get_xs ! Gets Size of Data w/in Object
+    procedure :: calc_f => nuclideiso_calc_f ! Calcs f given mu
+  end type NuclideIso
+
+!===============================================================================
+! NuclideAngle contains the base MGXS data for a nuclide specifically for
+! explicit angle-dependent weighted MGXS
+!===============================================================================
+
+  type, extends(NuclideMG) :: NuclideAngle
+
+    ! Microscopic cross sections. Dimensions are: (Npol, Nazi, Nl, Ng, Ng)
+    real(8), allocatable :: total(:,:,:)        ! total cross section
+    real(8), allocatable :: absorption(:,:,:)   ! absorption cross section
+    real(8), allocatable :: scatter(:,:,:,:,:)  ! scattering information
+    real(8), allocatable :: nu_fission(:,:,:,:) ! fission matrix (Gout x Gin)
+    real(8), allocatable :: k_fission(:,:,:)    ! kappa-fission
+    real(8), allocatable :: fission(:,:,:)      ! neutron production
+    real(8), allocatable :: chi(:,:,:)          ! Fission Spectra
+    real(8), allocatable :: mult(:,:,:,:)       ! Scatter multiplicity (Gout x Gin)
+
+    ! In all cases, right-most indices are theta, phi
+    integer              :: Npol         ! Number of polar angles
+    integer              :: Nazi         ! Number of azimuthal angles
```

Comment:
Can we call these `n_pol` and `n_azi`? This would match the style elsewhere and avoid the objectionable name of the second variable.